clients/gateway.go

```diff
@@ -37,10 +37,13 @@ func (c *GatewayClient) WithMinWait(d time.Duration) *GatewayClient {
 	return c
 }
 
-// https://cloud.google.com/iot/docs/how-tos/exponential-backoff
-func ExponentialBackoff(wait time.Duration) time.Duration {
+func ExponentialUpToOneMinBackoff(wait time.Duration) time.Duration {
 	time.Sleep(wait)
-	return wait * 2
+	wait *= 2
+	if wait > time.Minute {
```

Comment:
I think it makes sense to make this configurable. Block times are different on different networks, so even a minute could be too much for some network.

Reply:
I agree. However, it will require adding a `maxWait` parameter to the `Backoff` function type that will always be have the same value. Since we already have `WithBackoff`, do you think it is enough for users to implement their own?

Alternatively, we could export default functions for the common networks `MainnetBackoff`, `GoerliBackoff`, etc. that wrap a function like `ExponentialBackoffWithMaximum`. 

Ex:
```go
func MainnetBackoff(wait time.Duration) time.Duration {
        ExponentialBackoffWithMaximum(wait, time.Minute)
}
```

Reply:
How about;

``` golang
type Backoff func(numOfTriesSoFar uint)  time.Duration
```

and we can make `ExponentialBackoff` a member function of GatewayClient. So it has access to the configuration

Reply:
The proposed function signature looks good to me, but users wouldn't be able to create their own backoff functions if it's a method.